homeassistant/components/media_player/ziggo_mediabox_xl.py

```diff
@@ -0,0 +1,218 @@
+"""
+Support for interface with a Ziggo Mediabox XL.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.ziggo_mediabox_xl/
+"""
+import logging
+import socket
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, MediaPlayerDevice,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    SUPPORT_NEXT_TRACK, SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE,
+    SUPPORT_PLAY, SUPPORT_PAUSE)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, STATE_UNKNOWN,
+    STATE_PAUSED, STATE_PLAYING)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+KNOWN_DEVICES_KEY = 'ziggo_mediabox_xl_known_devices'
+
+SUPPORT_ZIGGO = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | \
+    SUPPORT_NEXT_TRACK | SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_SELECT_SOURCE | SUPPORT_PLAY
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ziggo Mediabox XL platform."""
+    known_devices = hass.data.get(KNOWN_DEVICES_KEY)
+    if known_devices is None:
+        known_devices = set()
+        hass.data[KNOWN_DEVICES_KEY] = known_devices
+
+    # Is this a manual configuration?
+    if config.get(CONF_HOST) is not None:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+    elif discovery_info is not None:
+        host = discovery_info.get('host')
+        name = discovery_info.get('name')
+    else:
+        _LOGGER.warning("Cannot determine device")
+        return
+
+    # Only add a device once, so discovered devices do not override manual
+    # config.
+    hosts = []
+    ip_addr = socket.gethostbyname(host)
+    if ip_addr not in known_devices:
+        sock = socket.socket()
+        try:
+            state = sock.connect_ex((ip_addr, 5900))
+        except socket.error:
+            _LOGGER.error("Couldn't connect to %s", ip_addr)
+        if state == 0:
+            hosts.append(ZiggoMediaboxXLDevice(ip_addr, name))
+            known_devices.add(ip_addr)
+        else:
+            _LOGGER.error("Can't connect to %s", host)
+    else:
+        _LOGGER.warning("Ignoring duplicate Ziggo Mediabox XL %s", host)
+    add_devices(hosts, True)
+
+
+class ZiggoMediaboxXLDevice(MediaPlayerDevice):
+    """Representation of a Ziggo Mediabox XL Device."""
+
+    def __init__(self, host, name):
+        """Initialize the device."""
+        # Generate a configuration for the Samsung library
+        self._host = host
+        self._port = {"state": 62137, "cmd": 5900}
+        self._name = name
+        self._state = STATE_UNKNOWN
```

Comment:
Default to `self._state = None`. Hass core will translate this to unknown when sending data to the frontend. We're trying to move away from platforms explicitly using `STATE_UNKNOWN`.

Reply:
Done